src/Aspire.Hosting.Keycloak/KeycloakResourceBuilderExtensions.cs

```diff
@@ -194,4 +197,55 @@ public static IResourceBuilder<KeycloakResource> WithRealmImport(
             importFullPath,
             defaultOwner: KeycloakContainerImageTags.ContainerUser);
     }
+
+    /// <summary>
+    /// Configures Keycloak to run behind a reverse proxy with TLS termination.
+    /// </summary>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="endpointName">The name of the endpoint to use for hostname configuration. If not specified, uses the primary HTTP endpoint.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <remarks>
+    /// This method configures Keycloak for deployment scenarios where TLS termination is handled by a reverse proxy
+    /// (such as Azure Container Apps). It sets the necessary environment variables to enable HTTP communication
+    /// and proper proxy header handling.
+    /// 
+    /// The following Keycloak environment variables are configured:
+    /// <list type="bullet">
+    /// <item><description><c>KC_HTTP_ENABLED=true</c> - Enables HTTP since the reverse proxy handles TLS termination</description></item>
+    /// <item><description><c>KC_PROXY_HEADERS=xforwarded</c> - Configures Keycloak to respect X-Forwarded headers from the reverse proxy</description></item>
+    /// <item><description><c>KC_HOSTNAME</c> - Sets the hostname to match the endpoint URL for proper URL generation</description></item>
+    /// </list>
+    /// 
+    /// <example>
+    /// Configure Keycloak for deployment behind a reverse proxy
+    /// <code lang="csharp">
+    /// var keycloak = builder.AddKeycloak("keycloak");
+    /// 
+    /// // For production deployment with reverse proxy (e.g., Azure Container Apps)
+    /// if (!builder.ExecutionContext.IsRunMode)
+    /// {
+    ///     keycloak.WithReverseProxy();
+    /// }
+    /// </code>
+    /// </example>
+    /// </remarks>
+    public static IResourceBuilder<KeycloakResource> WithReverseProxy(this IResourceBuilder<KeycloakResource> builder, string? endpointName = null)
```

Comment:
We should expose a `PublishWithReverseProxy` that does the check to avoid checking builder.ExecutionContext.IsRunMode